src/Components/Footer/Footer.tsx

```diff
@@ -0,0 +1,77 @@
+import { FilterBy } from '../../types/FilterBy';
+import { Todo } from '../../types/Todo';
+import { TodosFilter } from '../TodosFilter/TodosFilter';
+import { deleteTodo } from '../../api/todos';
+
+type Props = {
+  numberOfNotCompleted: number;
+  filteredBy: FilterBy;
+  setFilteredBy: (filterBy: FilterBy) => void;
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+  setErrorMessage: (arg: string) => void;
+  setFocusedInput: (arg: boolean) => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  numberOfNotCompleted,
+  filteredBy,
+  setFilteredBy,
+  todos,
+  setTodos,
+  setErrorMessage,
+  setFocusedInput,
+}) => {
+  const handleClearCompleted = () => {
+    setErrorMessage('');
+    setFocusedInput(false);
+
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    const deletePromises = completedTodos.map(todo => {
+      return deleteTodo(todo.id)
+        .then(() => ({ id: todo.id, success: true }))
+        .catch(() => {
+          setErrorMessage('Unable to delete a todo');
+
+          return { id: todo.id, success: false };
+        });
+    });
+
+    Promise.all(deletePromises)
```

Comment:
seems like you use Promise.all almost everywhere to update many todos at the same time. it's not a bad approach, but you can consider the solution above